src/agreement.rs

```diff
@@ -25,10 +27,13 @@ pub struct SecretKey(pub(crate) Scalar);
 // #[zeroize(drop)]
 pub struct SharedSecret(pub(crate) MontgomeryPoint);
 
-impl From<[u8; 32]> for PublicKey {
-    /// Given a byte array, construct a x25519 `PublicKey`.
-    fn from(bytes: [u8; 32]) -> PublicKey {
-        PublicKey(MontgomeryPoint(FieldElement::from_bytes(&bytes).unwrap()))
+impl TryFrom<[u8; 32]> for PublicKey {
+    type Error = crate::Error;
+    /// Given a byte array, construct a x25519 `PublicKey`. It may fail if some
+    /// underlying checks fail.
+    fn try_from(bytes: [u8; 32]) -> Result<Self> {
+        let elem = FieldElement::from_bytes(&bytes)?;
```

Comment:
Just a nit, but I'd prefer not abbreviating names.

Reply:
It is a C-ism. Still hard to drop the bad habits.